primitives/types/account_id.go

```diff
@@ -0,0 +1,58 @@
+package types
+
+import (
+	"bytes"
+	"errors"
+	"reflect"
+
+	sc "github.com/LimeChain/goscale"
+)
+
+var (
+	errorPubKeyNotSupported = errors.New("public key type not supported")
+)
+
+type ISigner interface {
+	sc.Encodable
```

Comment:
You can still pass any Encodable type which is not type safe, so I think it should be an interface constraint:

```go
type Signer interface {
	Ed25519Signer | Sr25519Signer | EcdsaSigner
}
```